core/test/qa_SchedulerMessages.cpp

```diff
@@ -445,6 +447,72 @@ const boost::ut::suite TopologyGraphTests = [] {
             // };
         }
     };
+
+    "UI constraints setting test"_test = [] {
+        gr::Graph testGraph(context->loader);
+        auto&     copy1 = testGraph.emplaceBlock("gr::testing::Copy<float32>", {});
+        auto&     copy2 = testGraph.emplaceBlock("gr::testing::Copy<float32>", {});
+
+        TestScheduler scheduler(std::move(testGraph));
+        auto          makeUiConstraints = [](float x, float y) { return gr::property_map{{"x", x}, {"y", y}}; };
+
+        {
+            // Setting ui_constraints property for all blocks, universal
+            sendMessage<Set>(scheduler.toScheduler, "" /* serviceName */, block::property::kStagedSetting /* endpoint */, {{"ui_constraints", makeUiConstraints(43, 7070)}} /* data  */);
+            // Setting ui_constraints property for one block
+            sendMessage<Set>(scheduler.toScheduler, copy1->uniqueName() /* serviceName */, block::property::kStagedSetting /* endpoint */, {{"ui_constraints", makeUiConstraints(42, 6)}} /* data  */);
+
+            waitForReply(scheduler.fromScheduler);
+        }
+
+        const auto replyCount = scheduler.fromScheduler.streamReader().available();
+        for (std::size_t replyIndex = 0UZ; replyIndex < replyCount; replyIndex++) {
+            const Message reply = getAndConsumeFirstReplyMessage(scheduler.fromScheduler);
+            std::println("Got a reply {}:\n{}", replyIndex, reply);
+        }
+
+        expect(copy1->settings().applyStagedParameters().forwardParameters.empty());
+        expect(copy2->settings().applyStagedParameters().forwardParameters.empty());
+
+        auto uiConstraintsFor = [](const auto& block) {
+            return std::visit(meta::overloaded{
```

Comment:
It's just a unit-test but: having only two options a simple if-else would be more efficient w.r.t. compile-time (and probably also runtime) performance.